### PR TITLE
Fix #22653: Port water height fixes to parkpatch system

### LINKED ISSUE #22653: Port #19740 water height fixes to .parkpatch files
#19740 had several fixes for water height to be patched on scenarios that were held off until #21189 was merged. These can now be applied via the `.parkpatch` files.

### PR DESCRIPTION
**Pull request by Harry-Hopkinson**
Port water height fixes from this PR https://github.com/OpenRCT2/OpenRCT2/pull/19740 into the new parkpath system for https://github.com/OpenRCT2/OpenRCT2/issues/22653

### CODE REVIEW

**Comment by Gymnasiast** [comment]
@HtotheTML Could you test this?

**Comment by Harry-Hopkinson** [comment]
Some of these fixes depend on https://github.com/OpenRCT2/OpenRCT2/issues/22655 and https://github.com/OpenRCT2/OpenRCT2/issues/22654 which I think are to be done in another PR.

**Comment by tupaschoal** [comment]
> Some of these fixes depend on #22655 and #22654, which I believe are to be addressed in another PR.

Do they? Because as far as I see, this is only applying water fixes, which are already supported.

**Comment by Harry-Hopkinson** [comment]
Most of the patches include a ```RemoveTileElementsAt``` function which I thought was to be added in the other PR's.

**Comment by tupaschoal** [comment]
> Most of the patches from the PR #19740 include a `RemoveTileElementsAt` function which I thought was to be added in the other PR's.

I mean: are the fixes here only visible/relevant when those two issues are solved or do they already add value? If the latter, then we can proceed, if the former, then we have to hold them until the other issues are tackled.

As for them being in other PRs or not, it depends on the size of the patch. We usually prefer smaller PRs because it makes for an easier/faster to review code and can then likely be merged faster.

**Comment by Harry-Hopkinson** [comment]
I am really sorry - I was testing this incorrectly. I forgot to manually copy the parkpatches over to the data folder of the compiled Openrct2 so I was not seeing any changes. After doing this I can confirm that this is working.

**Comment by Harry-Hopkinson** [comment]
These fixes work without the other PR's.

**Comment by Harry-Hopkinson** [comment]
Mystic Mountain

Screenshot_20240830_180251

Crumbly woods

Screenshot_20240830_180343

Hydro hills

Screenshot_20240830_180424

Geoffrey Gardens

Screenshot_20240830_180454

Botany Breakers

Screenshot_20240830_180621

Six Flags Magic Mountain

Screenshot_20240830_180701

**Comment by HtotheTML** [comment]
Is the Icy Adventures (formerly Antarctic - Ecological Salvage) adjustment in this PR as well?

**Comment by HtotheTML** [comment]
I assume as mentioned to me in #19740 the other aspects of these fixes, such as removing now submerged scenery and adjusting terrain color (ex. Botany Breakers) will be separate PRs?

File: data/scenario_patches/aead175.parkpatch [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+            "height": 96,
+            "coordinates": [
+                [39, 50], [39, 51], [39, 52], [39, 53], [39, 54], [39, 55], 
+                [40, 51], [40, 52], [40, 53], [40, 54], 
+                [41, 52], [41, 53], [41, 54], [41, 55], 
+                [42, 51], [42, 52], [42, 53], [42, 54], [42, 55], 
+                [43, 50], [43, 51], [43, 52], [43, 53], [43, 54], [43, 55], 
+                [44, 50], [44, 51], [44, 52], [44, 54], [44, 55], [44, 56], 
+                [45, 55], [45, 56], [45, 57], 
+                [46, 55], [46, 56], [46, 57]
```
Comment (tupaschoal) [inline data/scenario_patches/aead175.parkpatch:15]: Please update this one to use the same indenting as the "Botany Breakers (.sea)" file (`2ca68df.parkpatch`)

**Review by tupaschoal — COMMENTED** [review]
I have reviewed that all water fixes were ported correctly from the older PR and checked that the patches load and apply required fixes on all, but Botany Breakers and Icy Adventures (these are the bigger ones so I just didn't have time to check yet).

Btw @Harry-Hopkinson, I also rebased it to account for my recently merged patch.

File: data/scenario_patches/aead175.parkpatch (hunk 1 above)
Comment (Harry-Hopkinson) [reply data/scenario_patches/aead175.parkpatch]: Thanks for rebasing, I have updated the indenting of the Botany Breakers parkpatch.

**Comment by Harry-Hopkinson** [comment]
Would this need a changelog entry?

**Comment by tupaschoal** [comment]
> Would this need a changelog entry?

Yes!

**Comment by Harry-Hopkinson** [comment]
@tupaschoal Done.

**Review by tupaschoal — APPROVED** [review]
Checked remaining ones and it all seems good to go.

**Comment by Harry-Hopkinson** [comment]
Thanks for rebasing it for me 😄 , saved me a couple of minutes.

**Comment by tupaschoal** [comment]
> Thanks for rebasing it for me 😄 , saved me a couple of minutes.

Btw I fixed the changelog to point to the issue # instead of the PR #. We usually use PR# only when there is no tracking issue

File: distribution/changelog.txt [hunk 2]
```diff
@@ -1,5 +1,6 @@
 0.4.15 (in development)
 ------------------------------------------------------------------------
+- Fix: [#22653] Add several .parkpatch files for missing water tiles in RCT1 and RCT2 scenarios.
```
Comment (Gymnasiast) [inline distribution/changelog.txt:3]: These entries are incorrect in all parkpatch PRs. It should list the _problem_, and _not_ the fix.

**Pull request merged**